cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -718,6 +719,28 @@ public void liveInfo(HttpRequest request, HttpResponder responder, @PathParam("n
                 runtimeService);
   }
 
+  @GET
+  @Path("/apps/{app-id}/schedules/{program-type}/{program-name}")
+  public void programSchedules(HttpRequest request, HttpResponder responder,
```

Comment:
Added test case for the API. Yeah lets discuss about the CDAP-1073 and it can be done as separate PR.